assets/js/admin.js

```diff
@@ -37,24 +38,60 @@ jQuery(document).ready(function($) {
         media_frame.open();
     });
 
-    function configureHideShowAutoLogin() {
-        // Hide/Show login method depending on auto login
-        var $loginMethodField = $("#wpa0_auto_login_method").closest("tr");
-        var $autoLoginCheckbox = $("#wpa0_auto_login");
-        if (!$autoLoginCheckbox.prop("checked")) {
-            $loginMethodField.hide();
-        }
+    // Show/hide field for specific switches
+    $('[data-expand!=""]').each( function() {
+        var $thisSwitch = $( this );
+        var $showFieldRow = $( '#' + $thisSwitch.attr( 'data-expand' ) ).closest( 'tr' );
 
-        $autoLoginCheckbox.change(function() {
-            if (!$autoLoginCheckbox.prop("checked")) {
-                $loginMethodField.hide();
-            } else {
-                $loginMethodField.show();
+        if ( $showFieldRow.length ) {
+            if ( ! $thisSwitch.prop( 'checked' ) ) {
+                $showFieldRow.hide();
             }
+            $thisSwitch.change(function() {
+                if ( $( this ).prop( 'checked' ) ) {
+                    $showFieldRow.show();
+                } else {
+                    $showFieldRow.hide();
+                }
+            } );
+        }
+    });
 
-        });
+    // Persistent admin tab
+    if ( localStorageAvailable() ) {
+        window.location.hash = window.localStorage.getItem( 'Auth0WPSettingsTab' );
+        $( '.nav-tabs [role="tab"]' ).click( function () {
+            var tabHref = $( this ).attr( 'href' );
+            window.location.hash = tabHref;
+            window.localStorage.setItem( 'Auth0WPSettingsTab', tabHref );
+        } );
     }
 
-    configureHideShowAutoLogin();
+    // Clear cache button on Basic settings page
+    var deleteCacheId = 'auth0_delete_cache_transient';
+    var $deleteCacheButton = $( '#' + deleteCacheId );
+    $deleteCacheButton.click( function(e) {
+        e.preventDefault();
+        $deleteCacheButton.prop( 'disabled', true ).val( wpa0.clear_cache_working );
+        var postData = {
+            'action': deleteCacheId,
+            '_ajax_nonce': wpa0.clear_cache_nonce
+        };
 
+        $.post(wpa0.ajax_url, postData, function() {
+            $deleteCacheButton.prop( 'disabled', false ).val( wpa0.clear_cache_done );
+        }, 'json');
+    } );
+
+    function localStorageAvailable() {
+        try {
+            var x = '__storage_test__';
+            window.localStorage.setItem(x, x);
```

Comment:
is this try/catch the only way to check for availability? And, can the `removeItem` fail after the first one succeeded?

Reply:
No, but it's the best way:

https://developer.mozilla.org/en-US/docs/Web/API/Web_Storage_API/Using_the_Web_Storage_API#Feature-detecting_localStorage

As for `removeItem`, that would be a weird case, not implementing the webStorage interface. But localStorage persist forever and we don't want to store something we're never going to use.

Reply:
is this stored under some domain key or can it collide with existing "x" stored keys? In that case, you might want to use some key like "a0_local_storage_assertion"

Reply:
It's per-domain:

https://stackoverflow.com/q/4201239/728480

But you have a good point about that. WP sites could load a million other JS libraries.